taxonomy-editor-frontend/src/pages/startproject/index.jsx

```diff
@@ -0,0 +1,100 @@
+import { Typography, Box, Grid, TextField, Stack, Autocomplete, Snackbar, Alert } from "@mui/material";
+import LoadingButton from '@mui/lab/LoadingButton';
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { TAXONOMY_NAMES } from "../../constants";
+import { createBaseURL } from "../editentry/createURL";
+
+const StartProject = () => {
+    const [branchName, setBranchName] = useState("")
+    const [taxonomyName, setTaxonomyName] = useState("additives")
+    const [description, setDescription] = useState("");
+    const [loading, setLoading] = useState(false); 
+    const [open, setOpen] = useState(false);
+    const [errorMessage, setErrorMessage] = useState("")
+    const navigate = useNavigate();
+    
+    function handleSubmit() {
+        const url = createBaseURL(taxonomyName, branchName)
+        setLoading(true);
+        const dataToBeSent = {'description' : description};
+        fetch(url+'import', {
+            method : 'POST',
+            headers: {"Content-Type" : "application/json"},
+            body: JSON.stringify(dataToBeSent)
+        }).then((response) => {
+            if (!response.ok) {
+                return response.json();
+            }
+            navigate(`/${taxonomyName}/${branchName}/entry`)
+        }).then((responseBody) => {
+            setErrorMessage(responseBody.detail)
+            setOpen(true); setLoading(false);
+        })
+    }
+    function handleClose() {setOpen(false);}
+    return (
+        <Box>
+            <Grid
+            container
+            direction="column"
+            alignItems="center"
+            justifyContent="center"
+            >
+                <Typography sx={{mt: 4}} variant="h3">Start a project</Typography>
+                <Stack sx={{mt: 4, mb: 4}} direction="row" alignItems="center">
+                    <Typography sx={{mr: 4}} variant="h5">Taxonomy Name</Typography>
+                    <Autocomplete
+                        sx={{width: 265}}
+                        options={ TAXONOMY_NAMES }
+                        onChange={(e, selectedTaxonomy) => {
+                            if (selectedTaxonomy) setTaxonomyName(selectedTaxonomy.toLowerCase().replaceAll(/\s/g, '_'));
```

Comment:
is this a case for the functions you created for parsing the strings?

Reply:
Ah yeah, changed it to the function `toSnakeCase`